Dockerfile-auth

```diff
@@ -0,0 +1,39 @@
+# our chosen base image
+FROM debian:10
+
+# import everything - this could be pdns.git OR an auth tarball!
+COPY . /source
```

Comment:
Same applies to the other Dockerfiles.

Reply:
> Do this as late as possible, definitely _after_ the installation of all the build dependencies.

I currently steal the list of build dependencies, at build time, from our Debian sources. To move the COPY to later, I'd have to copy that list and then we'd be maintaining it in two places.. or perhaps I could copy just the data for the build deps here!

I should definitely have a `.dockerignore`.

As for figuring out a version number, I don't think a Dockerfile can decide what version number the image gets anyway, so that would have to happen externally I believe?

Reply:
Yes, you could just copy the the files you need at each stage.

For example, for Python projects I usually first add `requirements.txt` and `pip install` those, and only then add the rest of the sources. This way changing the sources does not trigger a reinstall of all third party packages.

Reply:
You could pass the version you want through an optional build arg, I'm just not sure if you can easily pass that to the rest of the build system.